lumicks/pylake/fitting/profile_likelihood.py

```diff
@@ -0,0 +1,333 @@
+from scipy.stats import chi2
+import numpy as np
+from typing import NamedTuple
+import matplotlib.pyplot as plt
+from warnings import warn
+
+
+def clamp_step(x_origin, x_step, lb, ub):
+    """Shortens a step to stay within some box constraints."""
+    alpha_ub = np.inf * np.ones(x_step.shape)
+    alpha_lb = -np.inf * np.ones(x_step.shape)
+
+    mask = x_step != 0
+    alpha_ub[mask] = (ub[mask] - x_origin[mask]) / x_step[mask]
+    alpha_lb[mask] = (lb[mask] - x_origin[mask]) / x_step[mask]
+
+    scaling = np.min(np.maximum(alpha_ub, alpha_lb))
+    if abs(scaling) > 1.0:
+        scaling = 1.0
```

Comment:
Actually, had to look twice, but it's the `abs` that's not needed here :)

I added a few comments, since upon revisiting after a while of not having seen it, the method wasn't immediately clear to me anymore.

As long as the initial point is inside the boundary (which it should be, I added an assert for this), scaling should always be positive as it reflects the distance towards each boundary.